File: app/src/main/java/com/example/rankers/kakao/GlobalApplication.java
package com.example.rankers.kakao;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.app.Application;
import android.content.Context;

import com.kakao.auth.ApprovalType;
import com.kakao.auth.AuthType;
import com.kakao.auth.IApplicationConfig;
import com.kakao.auth.ISessionConfig;
import com.kakao.auth.KakaoAdapter;
import com.kakao.auth.KakaoSDK;

public class GlobalApplication extends Application {

    private static GlobalApplication instance;

    // 깃 테스트 중입니다.

    public static GlobalApplication getGlobalApplicationContext() {
        if (instance == null) {
            throw new IllegalStateException("This Application does not inherit com.kakao.GlobalApplication");
        }
        return instance;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
        // Kakao Sdk 초기화
        KakaoSDK.init(new KakaoSDKAdapter());
    }
    @Override
    public void onTerminate() {
        super.onTerminate();
        instance = null;
    }

}

File: app/src/main/java/com/project/rankers/kakao/GlobalApplication.kt
package com.project.rankers.kakao

import android.app.Application

import com.kakao.auth.KakaoSDK

class GlobalApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        instance = this
        // Kakao Sdk 초기화
        KakaoSDK.init(KakaoSDKAdapter())
    }

    override fun onTerminate() {
        super.onTerminate()
        instance = null
    }

    companion object {

        private var instance: GlobalApplication? = null

        val globalApplicationContext: GlobalApplication
            get() {
                if (instance == null) {
                    throw IllegalStateException("This Application does not inherit com.kakao.GlobalApplication")
                }
                return instance as GlobalApplication
            }
    }

}